src/v2/Apps/Authentication/Utils/helpers.ts

```diff
@@ -143,9 +143,22 @@ export async function apiAuthWithRedirectUrl(
   response: Response,
   redirectPath: URL
 ): Promise<URL> {
-  const redirectUrl = sd.APP_URL + redirectPath.pathname + redirectPath.search
+  let redirectURL = redirectPath
+
+  const allowedHosts = sd.AUTH_REDIRECT_ALLOWED_HOSTS?.split(",") || []
```

Comment:
With this behavior, all hosts need to be configured (including the existing `APP_URL`). Alternatively, we could add logic to allow for both `AUTH_REDIRECT_ALLOWED_HOSTS` and `APP_URL`, which will also allow for a less risky deploy.

Reply:
It's reasonable to apply _different_ logic when redirecting internally vs. externally, so I think it would be nice if `AUTH_REDIRECT_ALLOWED_HOSTS` included only the external hosts, and `APP_URL` was accepted independently. (That also makes the empty default safer to deploy and reasonable for development.)